third_party/libdatachannel/BUILD.gn

```diff
@@ -0,0 +1,155 @@
+    "repo/src",
+    "repo/include/rtc",
+    "repo/deps/plog/include",
+    "repo/deps/usrsctp/usrsctplib",
+    "repo/deps/libjuice/include",
+    "repo/deps/libsrtp/include",
+  ]
+
+  # ---------------------------------------------------------------------------
+  # Source files from repo/src/*.cpp
+  # ---------------------------------------------------------------------------
+  sources = [
+    "repo/src/av1rtppacketizer.cpp",
+    "repo/src/candidate.cpp",
+    "repo/src/capi.cpp",
+    "repo/src/channel.cpp",
+    "repo/src/configuration.cpp",
+    "repo/src/datachannel.cpp",
+    "repo/src/description.cpp",
+    "repo/src/global.cpp",
+    "repo/src/h264rtpdepacketizer.cpp",
+    "repo/src/h264rtppacketizer.cpp",
+    "repo/src/h265nalunit.cpp",
+    "repo/src/h265rtpdepacketizer.cpp",
+    "repo/src/h265rtppacketizer.cpp",
+    "repo/src/mediahandler.cpp",
+    "repo/src/message.cpp",
+    "repo/src/nalunit.cpp",
+    "repo/src/pacinghandler.cpp",
+    "repo/src/peerconnection.cpp",
+    "repo/src/plihandler.cpp",
+    "repo/src/rembhandler.cpp",
+    "repo/src/rtcpnackresponder.cpp",
+    "repo/src/rtcpreceivingsession.cpp",
+    "repo/src/rtcpsrreporter.cpp",
+    "repo/src/rtp.cpp",
+    "repo/src/rtpdepacketizer.cpp",
+    "repo/src/rtppacketizationconfig.cpp",
+    "repo/src/rtppacketizer.cpp",
+    "repo/src/track.cpp",
+    "repo/src/websocket.cpp",
+    "repo/src/websocketserver.cpp",
+  ]
+
+  # ---------------------------------------------------------------------------
+  # Source files from repo/src/impl/*.cpp
+  # ---------------------------------------------------------------------------
+  sources += [
+    "repo/src/impl/certificate.cpp",
+    "repo/src/impl/channel.cpp",
+    "repo/src/impl/datachannel.cpp",
+    "repo/src/impl/dtlssrtptransport.cpp",
+    "repo/src/impl/dtlstransport.cpp",
+    "repo/src/impl/http.cpp",
+    "repo/src/impl/httpproxytransport.cpp",
+    "repo/src/impl/icetransport.cpp",
+    "repo/src/impl/init.cpp",
+    "repo/src/impl/logcounter.cpp",
+    "repo/src/impl/peerconnection.cpp",
+    "repo/src/impl/pollinterrupter.cpp",
+    "repo/src/impl/pollservice.cpp",
+    "repo/src/impl/processor.cpp",
+    "repo/src/impl/sctptransport.cpp",
+    "repo/src/impl/sha.cpp",
+    "repo/src/impl/tcpserver.cpp",
+    "repo/src/impl/tcptransport.cpp",
+    "repo/src/impl/threadpool.cpp",
+    "repo/src/impl/tls.cpp",
+    "repo/src/impl/tlstransport.cpp",
+    "repo/src/impl/track.cpp",
+    "repo/src/impl/transport.cpp",
+    "repo/src/impl/utils.cpp",
+    "repo/src/impl/verifiedtlstransport.cpp",
+    "repo/src/impl/websocket.cpp",
+    "repo/src/impl/websocketserver.cpp",
+    "repo/src/impl/wshandshake.cpp",
+    "repo/src/impl/wstransport.cpp",
+  ]
+
+  public_configs = [ ":libdatachannel_config" ]
+  configs += [
+    ":libdatachannel_config_disable_warnings",
+    ":libdatachannel_config_enable_features",
+  ]
+
+  defines = [
+    "BUILD_SHARED_LIBS=1",  # default ON in the CMake
```

Comment:
where are these defines used? Could we place them in a build_config header instead of on the command line of the compilation?

Reply:
Those defines are referenced in the source of repo libdatachannel, we can move those defines to build_config.h, but we need to include build_config.h wherever those macros are needed

Pros of using GN defines:
Makes it straightforward to override definitions at build time without editing source code (e.g., via command-line flags or GN args).

Keeps build-time toggles separate from the code (some prefer build-time config to remain in the build system, not in the repo code).